IM_calculation/Advanced_IM/check_adv_IM_status.py

```diff
@@ -0,0 +1,230 @@
+"""script to quickly check the status of adv_im for a event
+checks for existance of aggregated csv
+checks for nan/null in the csv
+checks station in csv matches stations ran
+"""
+
+
+import argparse
+from datetime import datetime
+from enum import Enum
+import os
+import sys
+
+import pandas as pd
+
+from qcore import constants as const
+from qcore.formats import load_station_file
+from IM_calculation.Advanced_IM.runlibs_2d import check_status, TIME_FORMAT, time_type
+
+
+class analysis_status(Enum):
+    not_started = 0
+    finished = 1
+    not_converged = 2
+    not_finished = 3
+    crashed = 4
+    unknown = 5
+
+
+COLUMN_NAMES = [
+    "station",
+    "model",
+    "component",
+    "status",
+    time_type.start_time.name,
+    time_type.end_time.name,
+]
+
+
+def read_timelog(comp_run_dir):
+    t_list = [None for y in time_type]
+    for t_type in time_type:
+        path_logfile = os.path.join(comp_run_dir, t_type.name)
+        if not os.path.isfile(path_logfile):
+            continue
+        with open(path_logfile, "r") as f:
+            time = datetime.strptime(f.readline(), TIME_FORMAT)
+            t_list[t_type.value] = time
+    return t_list
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "im_calc_dir", type=str, help="The path to the realisation directory"
+    )
+    parser.add_argument(
+        "adv_im_model", nargs="+", type=str, help="list of adv_IM models that ran"
+    )
+
+    parser.add_argument(
+        "--components",
+        default=[const.Components.c000.str_value, const.Components.c090.str_value],
+        nargs="+",
+        choices=list(const.Components.iterate_str_values()),
+        help="list of component that ran",
```

Comment:
components

Reply:
can i merge this PR and put this fix in the next one XD

Reply:
Yep. It's approved. =)